readthedocs/vcs_support/backends/git.py

```diff
@@ -144,7 +151,17 @@ def use_shallow_clone(self):
         return not self.project.has_feature(Feature.DONT_SHALLOW_CLONE)
 
     def fetch(self):
-        cmd = ['git', 'fetch', '--tags', '--prune', '--prune-tags']
+        cmd = ['git', 'fetch', 'origin',
+               '--tags', '--prune', '--prune-tags']
+
+        if (
+            self.verbose_name and
+            self.version_type == EXTERNAL and
+            'github.com' in self.repo_url
+        ):
+            cmd.append(
```

Comment:
I think this is better to be added after the next `if`. That way, all the attributes will be together and finally the "URL or the PR".